src/app/containers/PageHandlers/withData.jsx

```diff
@@ -1,31 +1,47 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import { shape, element } from 'prop-types';
 import articlePropTypes from '../../models/propTypes/article';
 import ErrorMain from '../ErrorMain';
 import nodeLogger from '../../helpers/logger.node';
 import deepGet from '../../helpers/json/deepGet';
+import { ServiceContext } from '../../contexts/ServiceContext';
+import getPassportHome from '../../lib/utilities/getPassportHome';
 
 const logger = nodeLogger(__filename);
 
-// checks for data and status keys, setting default status if not found
+// checks for data, status, setting default status if not found
 const constructRenderObject = data => ({
   status: deepGet(['status'], data) || 500,
   pageData: deepGet(['pageData'], data),
 });
 
-// checks for pageData and 200 status
+// checks for pageData, 200 status and if home service from article data fits the service locale
 const shouldRender = data => {
   const { status, pageData } = constructRenderObject(data);
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const { service } = useContext(ServiceContext);
 
   const hasDataAnd200Status = pageData && status === 200;
 
-  return { hasDataAnd200Status, status, pageData };
+  const passportHome = pageData && getPassportHome(pageData);
+
+  const isValidHome = pageData && passportHome === service;
+
+  return {
+    hasDataAnd200Status,
+    isValidHome,
+    status: hasDataAnd200Status && !isValidHome ? 404 : status,
```

Comment:
Just being nosey, but a heads up, as far as I can see, this wont work for WS frontpages which dont use the new optimo/ares data. An example of the data can be found [here](https://github.com/bbc/simorgh/blob/latest/data/prod/igbo/frontpage/index.json)

Could this logic be put somewhere else? maybe in a separate pageHandler after this one?

Reply:
Just checked and indeed the story for frontpages `http://localhost:9001/?path=/story/front-page--yoruba` now shows an error page. This should have caused a test to fail, flagging this earlier, so apologies for that. I'll look into writing a test to flag this.

Reply:
@dr3 thanks for pointing that out. I have updated the logic to work for WS frontpages as well. I think the logic is more suited to this HOC, because it checks if there's data, routes to the appropriate page and also displays the error page if an error is thrown